diff --git a/Makefile.am b/Makefile.am
index 89768e6..c9c3fbe 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -3352,7 +3352,9 @@ libsystemd_login_la_LDFLAGS = \
 	-Wl,--version-script=$(top_srcdir)/src/login/libsystemd-login.sym
 
 libsystemd_login_la_LIBADD = \
-	libsystemd-shared.la
+	libsystemd-shared.la \
+	libsystemd-daemon.la \
+	libsystemd-id128-internal.la
 
 libsystemd_login_internal_la_SOURCES = \
 	$(libsystemd_login_la_SOURCES)
